### PR TITLE
Tools:tune:src scripts does not execute with Matlab

### PR DESCRIPTION
**Pull request by ShriramShastry**
Running the  matlab script from 
```
\tool\tune\src\src_std_int32.m
\tool\tune\src\src_tiny_int16.m
\tool\tune\src\src_small_int32.m

```
is showing error in the command window
```
Error message 
Error using fprintf
Invalid file identifier. Use fopen to generate a valid file identifier.
Error in src_generate (line 171)
fprintf(fh,'\n');

Error in src_std_int32 (line 55)
src_generate(fs1, fs2, fs_matrix, cfg);
```

`Line fh = fopen(fn,'w')  - This is not executing , fh is -1`

### CODE REVIEW

**Comment by singalsu** [comment]
Should this be an issue and not a pull request? The patch looks like the relative paths add not related to description.

**Comment by lgirdwood** [comment]
@singalsu I guess this is all working for you ? If so, do we need a documentation update or some extra environment or $PATH settings here ? I think @ShriramShastry has run into this issue twice now with different tools.

**Comment by singalsu** [comment]
@lgirdwood I get with Matlab the same error but the patch doesn't look like it's a fix for this. The issue is weird. The fopen() fails for a text file create into subdirectory reports. In the beginning of this script there's a mkdir() done if the directory doesn't exist.

**Comment by singalsu** [comment]
@sriramshastry It looks like 'reports' is reserved work in Matlab. The check for `exist('reports','dir')` does not work correctly. Could you change the patch e.g. to:

```
diff --git a/tools/tune/src/src_generate.m b/tools/tune/src/src_generate.m
index c94ce4b73..446b515a1 100644
--- a/tools/tune/src/src_generate.m
+++ b/tools/tune/src/src_generate.m
@@ -73,7 +73,7 @@ end
 data_bytes = 4;
 
 hdir = mkdir_check('include');
-rdir = mkdir_check('reports');
+rdir = mkdir_check('logs');
 
 %% Find fractional conversion factors
 nfsi = length(fs_in);

```

This changes the 'reports' directory name to 'logs'. It's a silly change but I can't think of anything else. With name 'reports' it works if a shell command `mkdir reports` is run before this Matlab script.

**Comment by lgirdwood** [comment]
@ShriramShastry does this work for you now ? It sounds like ```reports``` may be a reserved word ?

**Comment by ShriramShastry** [comment]
> @ShriramShastry does this work for you now ? It sounds like `reports` may be a reserved word ?

I have not verified up to now. I 'II check and update the feedback(Sorry about that)

**Comment by lgirdwood** [comment]
@ShriramShastry any update ?

**Comment by ShriramShastry** [comment]
@lgirdwood , Sorry for the delay
I'II update my feedback by  Friday EOD.

**Comment by ShriramShastry** [comment]
isfolder can be used instead of exist('reports','dir'), exist checks complete path

```
```
I made a sample code to understand the why report is not working 

function src_gen()
hdir = mkdir_check('include');
rdir = mkdir_check('reports');
end
function d = mkdir_check(d)
if exist(d) ~= 7
    mkdir(d);
end
end

Command window result for d 
K>> exist(d)      % This is  for include 
     0

K>> exist(d)      % This is for  reports
     7

```
```
For  reports, d returns 7 and check is  for ~=7. So it is not working

**Comment by lgirdwood** [comment]
@ShriramShastry thanks, @singalsu do you agree ?

**Review by singalsu — CHANGES_REQUESTED** [review]
Just to make sure, a reject, the code PR here is not a proper fix.

**Comment by singalsu** [comment]
Yep, I agree with @ShriramShastry 's test done above. Though my Matlab version is R2017a, and it doesn't have isfolder(). Isfolder is available in Octave so OK for me to change. Can you please update this this PR to contain code that addresses the issue!

**Comment by lgirdwood** [comment]
@ShriramShastry do you have a proper fix pending ?

**Comment by ShriramShastry** [comment]
Yes- I will make change for  proper fix!

**Review by ShriramShastry — COMMENTED** [review]
Audio : Tools : src generate folder check fails 
Replace exist function check with isfolder function check. isfolder fcn return logical value unlike exist checks for size of char

File: tools/tune/src/src_generate.m
Comment (singalsu) [inline tools/tune/src/src_generate.m:242]: Looks OK but add a description of problem you fixed into commit message. Also your full name as  you prefer to write it to signed-off-by please. I got recently my signoff messed up similarly due to another Intel github project configured.

**Review by singalsu — CHANGES_REQUESTED** [review]

File: tools/tune/src/src_generate.m
Comment (singalsu) [inline tools/tune/src/src_generate.m:245]: One more thing, https://se.mathworks.com/help/matlab/ref/isfolder.html

isfolder is never 7 so this should be `if ~isfolder(d)`, right?

Comment (ShriramShastry) [reply tools/tune/src/src_generate.m:245]: isfolder result is logical 0/1 but ~=7 impact is seen when Exporting the file to include folder in case ~=7 is removed
so  left ~=7 as it is
```
>> src_std_int32
Step=  1, Delta=36.133008 dB, N=131
Step=  2, Delta=34.465830 dB, N=135
Step=  3, Delta=32.693937 dB, N=139
Step=  4, Delta=30.804600 dB, N=143
Step=  5, Delta=28.776715 dB, N=147
Step=  6, Delta=26.587632 dB, N=151
Step=  7, Delta=24.200499 dB, N=155
Step=  8, Delta=21.568589 dB, N=159
Step=  9, Delta=18.614211 dB, N=163
Step= 10, Delta=15.215722 dB, N=167
Step= 11, Delta=11.139196 dB, N=171
Step= 12, Delta=5.862646 dB, N=175
Step= 13, Delta=0.263999 dB, N=179
Step= 14, Delta=0.277000 dB, N=183
Step= 15, Delta=0.276205 dB, N=187
Step= 16, Delta=0.032672 dB, N=191
Step= 17, Delta=0.107534 dB, N=195
Step= 18, Delta=-0.191142 dB, N=199
Exporting include/src_std_int32_1_2_4535_5000.h ...
error: fprintf: invalid stream number = -1
error: called from
    src_export_coef at line 43 column 2
    src_generate at line 156 column 25
    src_std_int32 at line 55 column 1
```

File: tools/tune/src/src_generate.m
Comment (ShriramShastry) [reply tools/tune/src/src_generate.m:242]: OK, I will add the description.

File: tools/tune/src/src_generate.m
Comment (singalsu) [reply tools/tune/src/src_generate.m:245]: Doesn't the always true if mean that mkdir() is executed always? I added this because in matlab or octave there was unnecessary warnings printed to screen if the directory existed.

Comment (ShriramShastry) [reply tools/tune/src/src_generate.m:245]: Yes mkdir() is  executed 
First when is created for first time , we don't have warning
and next time onwards ( when it already exist), we see the warnings 


```
Warning: Directory already exists. 
> In src_generate>mkdir_check (line 246)
  In src_generate (line 75)
  In src_std_int32 (line 55) 
```

I see the error due to cascaded affect as seen in below error logs  

```
Exporting include/src_std_int32_1_2_4535_5000.h ...
Error using fprintf
Invalid file identifier. Use fopen to generate a valid file identifier.

Error in src_export_coef (line 43)
	fprintf(fh, '/* SPDX-License-Identifier: BSD-3-Clause\n');

Error in src_generate (line 156)
                        src_export_coef(src1, coef_label, coef_ctype, hdir, cfg.profile);

Error in src_std_int32 (line 55)
src_generate(fs1, fs2, fs_matrix, cfg);
 
>> 

```

**Comment by lgirdwood** [comment]
@singalsu @ShriramShastry I cant add much value here since it's matlab/octave, so I will merge once you guys have aligned.

**Comment by singalsu** [comment]
The fix you proposed does not work in Octave correctly. If edit the function to be like this

```
function d = mkdir_check(d)
if isfolder(d) ~=7 % true if PATH points to a folder
	fprintf(1, 'Do mkdir %s\n', d);
        mkdir(d);
else
	fprintf(1, 'Directory %s exists\n', d);
end
end

```

And I make sure there's the directories the script does the mkdir every time:

```
>> ls -l
total 88
drwxrwxr-x 2 singalsu singalsu 4096 Jan  7 16:21 include
drwxrwxr-x 2 singalsu singalsu 4096 Nov 19 15:32 logs
-rw-rw-r-- 1 singalsu singalsu 1240 Sep 15 14:27 README
drwxrwxr-x 2 singalsu singalsu 4096 Jan  7 16:21 reports
-rw-rw-r-- 1 singalsu singalsu 7233 Sep 15 14:27 src_export_coef.m
-rw-rw-r-- 1 singalsu singalsu 1907 Sep 15 14:27 src_export_defines.m
-rw-rw-r-- 1 singalsu singalsu 5761 Sep 15 14:27 src_export_table_2s.m
-rw-rw-r-- 1 singalsu singalsu 1878 Sep 15 14:27 src_factor1_lm.m
-rw-rw-r-- 1 singalsu singalsu 4854 Sep 15 14:27 src_factor2_lm.m
-rw-rw-r-- 1 singalsu singalsu 2261 Sep 15 14:27 src_find_l0m0.m
-rw-rw-r-- 1 singalsu singalsu 9107 Jan  7 16:20 src_generate.m
-rw-rw-r-- 1 singalsu singalsu 7503 Sep 15 14:27 src_get.m
-rw-rw-r-- 1 singalsu singalsu 3122 Sep 15 14:27 src_param.m
-rw-rw-r-- 1 singalsu singalsu  624 Sep 15 14:27 src_small_int32.m
-rw-rw-r-- 1 singalsu singalsu 2365 Sep 15 14:27 src_std_int32.m
-rw-rw-r-- 1 singalsu singalsu 1966 Sep 15 14:27 src_tiny_int16.m
>> src_std_int32
Do mkdir include
Do mkdir reports

```

I didn't yet update my Matlab (a pretty time taking task), I'm still running  R2017a. It doesn't support isfolder(). But I get the same error as you if I remove the directories. I'd propose to use fullfile() to pre-pend the path to exist.

```
function d = mkdir_check(d)
if exist(fullfile('./', d)) ~=7 % true if PATH points to a folder
	fprintf(1, 'Do mkdir %s\n', d);
        mkdir(d);
else
	fprintf(1, 'Directory %s exists\n', d);
end
end

```
Does this work in your Matlab version?

**Comment by singalsu** [comment]
Actually `if exist(fullfile('.', d)) ~=7` is better since it avoids the UNIX / vs. Windows \ issue. Fullfile() handles it correctly for both.

File: tools/test/audio/std_utils/stdnotch_get.m
Comment (singalsu) [inline tools/test/audio/std_utils/stdnotch_get.m:46]: Unfortunately this is not sufficient, it should be e.g. fullfile('.', ' iirnotch.m'). When I tried, just s = fullfile('iirnotch.m') there's no path added to beginning.

File: tools/test/audio/test_utils/delete_check.m
Comment (singalsu) [inline tools/test/audio/test_utils/delete_check.m:33]: There's . missing from fullfile() I think

File: tools/test/audio/test_utils/load_test_output.m
Comment (singalsu) [inline tools/test/audio/test_utils/load_test_output.m:34]: The . is missing from here too

File: tools/test/audio/test_utils/mkdir_check.m
Comment (singalsu) [inline tools/test/audio/test_utils/mkdir_check.m:32]: Nitpick, in C code style we add one blank after commas, maybe we should in Matlab too. Though I'm not always doing it always myself.

**Review by singalsu — CHANGES_REQUESTED** [review]
I think some of the fixes would not work in Matlab due to not using '.' for fullfile path when the file name is like 'reports' that triggered this problem.

File: tools/test/audio/test_utils/delete_check.m
Comment (ShriramShastry) [reply tools/test/audio/test_utils/delete_check.m:33]: I'II push the code as per your suggestion but i am not sure of correctness of results.  When I debugged really and f output. The behavior was unexpected and  we see errors upon running the src_test(32,32)

File: tools/tune/eq/mls_freq_resp.m
Comment (singalsu) [inline tools/tune/eq/mls_freq_resp.m:276]: Should this be without last "> 0" ?

Comment (ShriramShastry) [reply tools/tune/eq/mls_freq_resp.m:276]: This is check for not empty . i.e. length higher than 0. So replace length with function call not empty

Comment (singalsu) [reply tools/tune/eq/mls_freq_resp.m:276]: Isn't just `if ~isempty(rec.cal)` enough?

**Review by singalsu — APPROVED** [review]
Looks OK now, thanks!

File: tools/tune/src/src_generate.m
Comment (cujomalainey) [reply tools/tune/src/src_generate.m:242]: Thanks for adding it, but please follow git format, first line is a very short summary, then a space, then a full body description

**Comment by ShriramShastry** [comment]
I am I allowed to merge to master branch or Liam can do it?
Thank you!

**Comment by singalsu** [comment]
> I am I allowed to merge to master branch or Liam can do it?
> Thank you!

Maintainers (e.g. Liam, Curtis, Daniel)  will do it for us after required number of approvals (2) and all green CI.

**Comment by singalsu** [comment]
@ShriramShastry The first CI issue is about your commit text format, please fix it.

File: tools/test/audio/std_utils/stdnotch_get.m
Comment (lgirdwood) [inline tools/test/audio/std_utils/stdnotch_get.m:46]: Looks like the commit subject overflowed and subject should be lowercase. Can you copy the message fix below.
```
audio : tools :  fix matlab/octave directory compatibility

To support exist function - fullfile() path addition is to prevent Matlab (differently than Octave)
to give false positive with some other directory location with it's search path

Added correction to isempty.

Signed-off-by: sriram <malladi.sastry@intel.com>
```

**Review by lgirdwood — CHANGES_REQUESTED** [review]

File: tools/test/audio/std_utils/stdnotch_get.m
Comment (ShriramShastry) [reply tools/test/audio/std_utils/stdnotch_get.m:46]: done

Comment (lgirdwood) [reply tools/test/audio/std_utils/stdnotch_get.m:46]: Have you pushed ? I can only see the old version ?

Comment (lgirdwood) [reply tools/test/audio/std_utils/stdnotch_get.m:46]: @ShriramShastry sorry, still seeing the old version ? fwiw ```git rebase -i``` can be used to fix this commit message, just mark it for reword(r) and edit the message and then you will need to force push otherwise the update wont be visible here.

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**
